python-threatexchange/threatexchange/exchanges/impl/ncmec_api.py

```diff
@@ -303,12 +310,25 @@ def log(event: str) -> None:
                         # Our entry estimatation (based on the cursor parameters)
                         # occasionally seem to over-estimate
                         log(f"est {entry.estimated_entries_in_range} entries")
-                elif i % 100 == 0:
+
+                updates.extend(entry.updates)
+
+                if i % 100 == 0:
                     # If we get down to one second, we can potentially be
                     # fetching an arbitrary large amount of data in one go,
-                    # so log something occasionally
-                    log(f"large fetch ({i}), up to {len(updates)}")
-                updates.extend(entry.updates)
+                    # so store the checkpoint occasionally
+                    log(f"large fetch ({i}), up to {len(updates)}. storing checkpoint")
```

Comment:
nit: You don't actually store the checkpoint by yielding, technically the caller can decide whether to keep calling or store.

Reply:
ah so the original `elif` block doesn't need to change? the only real change that's needed is to use the next_url in the for loop on L283?

Reply:
updated the comment 👍